Day 23: “The many lives of the page title” / Katrin Kampfrath (#214)
matuzo/HTMHell: hell/adventcalendar/2025/23/index.md

date: 2025-12-23
author_links:
  - label: "Blog"
    url: "https://frontend.die-katrin.eu/"
    link_label: "frontend.die-katrin.eu"
  - label: "Mastodon"
    url: "https://front-end.social/@katrin_k"
    link_label: "@katrin_k"
intro: "<p>A page title describes or identifies a page. So far, so good. But when and how is a page title actually relevant? Let's put on the user perspective hat and explore.</p>"
image: "advent25_23"
---

Along with its co-star, the favicon, it appears on stage in the browser's tab. It is presented on search result lists. It is part of the announcements of a screen reader when a page loads. The browser uses it as the proposed name of a bookmark: The page title.

It lives in its `<title>` tag within in the `<head>` tag, supposedly unseen and often overlooked. Most tips & tricks around the page title will revolve around SEO benefits. But that's not all. It plays a vital role in the user flow of navigating the web. 

> ### The user experience of a page title

Reviewer: This section presents the context really nicely, but I found the flow a bit hard to follow because the paragraphs break in places where the thought doesn't feel fully finished yet. 

I _think_ the narrative we're after here is 
* Broad problem statement (first para) – navigating between links makes it hard to reorient ourselves
* How the page title helps, with different contextual examples – screen reader users, sighted users, switching between windows, etc, → all leading to the conclusion that the page title reassures us we're in the right place.

Perhaps just combining them like this would feel smoother?

```
Clicking a link and having a new page load is the defining usage pattern of the web. But how do we know we've clicked the right link? How do we know which tab to return to when we've switched to another program?

For screen reader users the page title is among the first things announced. On page load, sighted users will most likely first notice the \`H1\` heading. When coming from another program or another browser tab, the page title helps us reorient ourselves. As such, a page title reassures us that we are indeed on the right page.
```

<img src="./bike-insurance.png" width="740" height="569" loading="lazy" alt="Browser tab for entering the bike details mentioned in the previous paragraph. In the center there are the form fields, on the left side is a navigation for the multiple steps, the browser tab shows the page title 'Buy bike insurance step 2'.">

The user navigates through the form. They fill in their name and address, click "next", now the bike details. Uff, what was the exact price? They might need to open an email client which means opening another program. They see another email, click on a link there, open another tab, etc.

Coming back to the form should now be as seamless as possible. When returning to a tab, screen readers announce at least the program and the page title of the current tab (the rest depends a little on the screen reader). Consider these two possible titles:
* "Bike details step 2 of 4 - Buy bike insurance - company name"
* "Buy bike insurance - company name"

> Example one is very precise, it leads the user directly back to where they left off. Example two sufficiently distinguishes the form from other content on the website. 

Reviewer: ```suggestion
Example 1 is very precise, leading the user directly back to where they left off. Example 2 sufficiently distinguishes the form from other content on the website. 
```

> Example one is very precise, it leads the user directly back to where they left off. Example two sufficiently distinguishes the form from other content on the website. 

Reviewer: I wasn’t fully sure which title you recommend — Option 1 as the ideal, and Option 2 as the minimum? The `sufficiently distinguishes` reads a bit neutral, maybe a small clarification here would help it land more clear.

```suggestion
When they return to the insurance form, it should be as seamless as possible. Screen readers will announce the program and the page title of the current tab (the rest depends a little on the screen reader). Clear page titles make it easy to identify where they left off.

Consider these two possible titles:

1. "Bike details - Step 2 of 4 · Buy bike insurance - Company Name"
2. "Buy bike insurance - Company Name"

The first one tells the user exactly where they are in the process. The second still identifies the form, but doesn’t reinforce their progress.
```

### Relevant WCAG requirements

What i mentioned so far concerns both accessibility and user experience aspects. Next i want to make more clear what is needed from an accessibility compliance perspective. 

> In the Web Content Accessibility Guidelines (WCAG) there are two related success criteria. Number one is [Success Criterion 2.4.2 Page Titled](https://www.w3.org/TR/WCAG22/#page-titled). It checks if a page title describes the topic or purpose of that current page. And there is [Success Criterion 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG22/#link-purpose-in-context). It concerns the links on a page.

Reviewer: ```suggestion
In the Web Content Accessibility Guidelines (WCAG) there are two related success criteria:

1. [Success Criterion 2.4.2 Page Titled](https://www.w3.org/TR/WCAG22/#page-titled) checks whether a page title describes the topic or purpose of the current page.
2. [Success Criterion 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG22/#link-purpose-in-context) concerns the links on a page.
```

With regard to accessibility and legal requirements, the WCAG (also included in the European Guideline EN 301549) states minimum requirements. Think of the color contrast of text to background. The contrast ratio of 4.5:1 is the minimum, but a stronger contrast is always better. 

Going back to our bike insurance example, this is how we would audit this process at our company. For success criterion 2.4.2 Page Titled we check:
1. Does each step have a unique URL? Then this page needs to have "Bike details" or "Bike details step 2 of 4" as part of its page title.
2. If there are no unique URLs, i.e. the form always has to be started at step 1, the overall form process must be sufficiently reflected in the page title. "Buy bike insurance - company name" would suffice for passing the success criterion. We would then add a recommendation to add the current step for user experience reasons. 

> For success criterion 2.4.4 Link Purpose (In Context) things are little less clear. Remember the typical flow of read a link, click a link, have a new page load. The main intent for the success criterion is that "[Users understand what each link will do](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html)". The criterion focuses on the side of "read a link" and as such an accessibility audit will check the link texts on a given page. 

Reviewer: ```suggestion
For Success Criterion 2.4.4 Link Purpose (In Context), things are a little less clear. Recall the typical flow: read a link, click a link, have a new page load. The main intent of the success criterion is that "[users understand what each link will do](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html)." The criterion focuses on the "read a link" part, so an accessibility audit will check the link texts on a given page.
```

Nonetheless, the user flow does play a role. In the WCAG's understanding document for success criterion 2.4.4 Link Purpose (In Context) it says:

> > Having the link and the title agree, or be very similar, is good practice and provides continuity between the link 'clicked on' and the web page that the user lands on.

Reviewer: The section feels like it ends a bit suddenly. Maybe add a short concluding sentence to pull it all together? Something that sums up "WCAG is the minimum, and adding step information helps all users, not only AT users." Right now it's a bit abrupt.

> > Having the link and the title agree, or be very similar, is good practice and provides continuity between the link 'clicked on' and the web page that the user lands on.

Reviewer: One possible ending, feel free to ignore!

```suggestion
> Having the link and the title agree, or be very similar, is good practice and provides continuity between the link 'clicked on' and the web page that the user lands on.

In other words: WCAG sets the minimum. Adding context about the current step in a flow helps not only assistive-technology users, but all users stay oriented.
```

## Conclusion

> Give your page sufficiently unique page titles. What is sufficient, depends on the implementation details of your website. 

Reviewer: ```suggestion
Give your page sufficiently unique page titles. What is sufficient depends on the implementation and design details of your website. 
```

> Give your page sufficiently unique page titles. What is sufficient, depends on the implementation details of your website. 

Reviewer 2: It depends on the specific use case, yes, but as a reader I'm now wondering which other example there are besides multi-step forms. You did such a great job outlining the importance of the attribute at the beginning and at the end, but in the middle it focuses too much on a specific use case. Readers shouldn't take away “Now I know how to add titles to multi step forms" but rather "Now I know that giving context in titles is important”. For that, the articles needs more examples. You don't have to describe them as detailed, but it would be great to see some more, maybe even based on real projects you were working on.

Here are three example I've used in my book “Web accessibility Cookbook”.

Example 1-18. Number of validation errors in the title of a sign up page.

```html
<title>2 errors - Sign Up - Johanna’s Toy Store</title>
```

Example 1-19. Number of results in a search page’s title.

```html
<title>21 results for term “crocodile” - Johanna’s Toy Store</title>
```
Example 1-20. The title indicating the current search results page.

```html
<title>Page 2 - Product Search Results - Johanna’s Toy Store</title>
```

Keep the user flow in mind. The page title is a key element to reassure users they are indeed on the page they intended to be. 

> Developer experience is not the same as user experience. As developers we know how things work on a technical level. We might easily forget how users might perceive a website. Also, never assume people will visit your site only in single straight-forward sessions with no distraction. 

Reviewer: I like the succinct, clear conclusion, but my only small note is that the original final sentence started with "Also", which made it feel like a new idea was being introduced at the end rather than bringing everything together. Maybe add a closing thought to bring it all together and finish strong?

```suggestion
Developer experience is not the same as user experience. As developers we know how things work on a technical level, and we might easily forget how users perceive a website. Never assume people will visit your site only in one straightforward session with no distractions. Clear, contextual titles help people stay oriented when they return.
```
